### PR TITLE
[Integration Tests] Cleanup usage of TestHelpers Part 1

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

**Comment by bokobza** [comment]
👀

File: src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public class CoreNode
         private readonly NetworkCredential creds;
         private readonly object lockObject = new object();
         private readonly ILoggerFactory loggerFactory;
-        private readonly NodeRunner runner;
+        internal readonly NodeRunner Runner;
```
[the remark this answers is not in the document]
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs]: Fixed

File: src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs [hunk 2]
```diff
@@ -331,7 +337,10 @@ public static void ConnectAndSync(CoreNode thisNode, params CoreNode[] to)
         /// <returns>Returns <c>true</c> if the address exists in this node's connected peers collection.</returns>
         public static bool IsNodeConnectedTo(CoreNode thisNode, CoreNode isConnectedToNode)
         {
-            return thisNode.FullNode.ConnectionManager.ConnectedPeers.Any(p => p.PeerEndPoint.Equals(isConnectedToNode.Endpoint));
+            if (thisNode.Runner is BitcoinCoreRunner)
```
[the remark this answers is not in the document]
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs]: I have fixed this.

File: src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs [hunk 3]
```diff
@@ -221,7 +227,7 @@ public static List<Transaction> Reorder(List<Transaction> transactions)
                 return transactions;
 
             var result = new List<Transaction>();
-            Dictionary<uint256, TransactionNode> dictionary = transactions.ToDictionary(t => t.GetHash(), t => new TransactionNode(t));
+            var dictionary = transactions.ToDictionary(t => t.GetHash(), t => new TransactionNode(t));
```
[the remark this answers is not in the document]
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs]: Fixed

File: src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs [hunk 4]
```diff
@@ -266,31 +227,27 @@ public void PullerVsMinerRaceCondition()
...
                 // Wait for block repository for block sync to work.
-                stratisMinerLocal.CreateRPCClient().AddNode(stratisMinerRemote.Endpoint, true);
-
-                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(stratisMinerLocal, stratisMinerRemote));
+                TestHelper.ConnectAndSync(stratisMinerLocal, stratisMinerRemote);
 
                 // Now disconnect the peers and mine block C2p on remote.
-                stratisMinerLocal.CreateRPCClient().RemoveNode(stratisMinerRemote.Endpoint);
-                TestHelper.WaitLoop(() => !TestHelper.IsNodeConnected(stratisMinerRemote));
+                TestHelper.ConnectAndSync(stratisMinerLocal, stratisMinerRemote);
```
[the remark this answers is not in the document]
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs]: Oops!

**Comment by bokobza** [comment]
Better build here: https://dev.azure.com/StratisProject/StratisBitcoinFullNode/_build/results?buildId=1004&view=ms.vss-test-web.test-result-details

**Pull request merged**